src/plugins/noPendingCount.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Vencord, a modification for Discord's desktop app
+ * Copyright (c) 2023 Vendicated and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { findByPropsLazy } from "@webpack";
+import { definePluginSettings } from "@api/Settings";
+
+const MessageRequestStore = findByPropsLazy("getMessageRequestsCount");
+
+const settings = definePluginSettings({
+    hideFriendRequestsCount: {
+        type: OptionType.BOOLEAN,
+        description: "Hide incoming friend requests count",
+        default: true,
+        restartNeeded: true
+    },
+    hideMessageRequestsCount: {
+        type: OptionType.BOOLEAN,
+        description: "Hide message requests count",
+        default: true,
+        restartNeeded: true
+    },
+    hidePremiumOffersCount: {
+        type: OptionType.BOOLEAN,
+        description: "Hide nitro offers count",
+        default: true,
+        restartNeeded: true
+    }
+});
+
+export default definePlugin({
+    name: "NoPendingCount",
+    description: "Removes the ping count of incoming friend requests, message requests, and nitro offers.",
+    authors: [Devs.amia],
+
+    settings: settings,
+
+    // Functions used to determine the top left count indicator can be found in the single module that calls getUnacknowledgedOffers(...)
+    // or by searching for "showProgressBadge:"
+    patches: [
+        {
+            find: ".getPendingCount=",
+            predicate: () => settings.store.hideFriendRequestsCount,
+            replacement: {
+                match: /(\i)\.getPendingCount=function\(\){return\s\i};/,
```

Comment:
```suggestion
                match: /\.getPendingCount=function\(\)\{/,
```

you can make this patch simpler by not unnecessarily matching \i and the function body and also using a lookbehind. Now your replacement only has to be `.getPendingCount=function(){return 0;`

You could make it even simpler by using a lookbehind:
```suggestion
                match: /(?<=\.getPendingCount=function\(\)\{)/,
```
Now your replacement becomes `return 0;`

Reply:
ah okay, for some reason I thought that causing the function to have double `return` would be bad

Reply:
no it's completely fine, we do it a lot as it makes many things a lot simpler